src/NServiceBus.Core/Diagnostics/ActivityDecorator.cs

```diff
@@ -15,40 +14,64 @@ public static void PromoteHeadersToTags(Activity activity, Dictionary<string, st
 
             foreach (var header in headers)
             {
-                if (header.Key.StartsWith("NServiceBus.") && !IgnoreHeaders.Contains(header.Key))
+                if (HeaderMapping.TryGetValue(header.Key, out var tagName))
                 {
-                    activity.AddTag(OtNamingConvention(header.Key), header.Value);
+                    activity.AddTag(tagName, header.Value);
                 }
             }
-
-            //TODO might be faster to just provide a hardcoded lookup if we're going with an allow-list approach
-            string OtNamingConvention(string pascalCasedString)
-            {
-                // use "nservicebus" instead of "n_service_bus"
-                pascalCasedString = pascalCasedString.Replace("NServiceBus", "nservicebus");
-                var additionalLength =
-                    pascalCasedString.Count(char.IsUpper) - pascalCasedString.Count(c => c == '.');
-                var result = new char[pascalCasedString.Length + additionalLength];
-                int i = 0;
-                foreach (char c in pascalCasedString)
-                {
-                    if (char.IsUpper(c) && i > 0 && result[i - 1] != '.')
-                    {
-                        result[i] = '_';
-                        i++;
-                    }
-
-                    result[i] = char.ToLower(c);
-                    i++;
-                }
-
-                return new string(result);
-            }
         }
 
-
-
-        // List of message headers that shouldn't be added as activity tags
-        static readonly HashSet<string> IgnoreHeaders = new HashSet<string> { Headers.TimeSent };
+        internal static readonly Dictionary<string, string> HeaderMapping = new()
+        {
+            //{ Headers.HttpFrom, "" },
```

Comment:
Are these to be cleaned up?

Reply:
yeah I'll remove them, I just wanted to show which one I opted-out from